File: problems/0146.LRU Cache/Solution.go
type Node struct {

	key, val   int
	prev, next *Node

}

type LRUCache struct {

	size, capacity int
	head, tail     *Node
	cache          map[int]*Node

}

func Constructor(capacity int) LRUCache {
    
}

func (this *LRUCache) Get(key int) int {
    
}

func (this *LRUCache) Put(key int, value int) {
    
}

func (this *LRUCache) removeNode(node *Node) {
    
}

func (this *LRUCache) addToHead(node *Node) {
    
}

/**
 * Your LRUCache object will be instantiated and called as such:
 * obj := Constructor(capacity);
 * param_1 := obj.Get(key);
 * obj.Put(key,value);
 */

/**
 * Your LRUCache object will be instantiated and called as such:
 * obj := Constructor(capacity);
 * param_1 := obj.Get(key);
 * obj.Put(key,value);
 */

File: problems/0146.LRU Cache/Solution.ts
class Node {

    key: number;
    val: number;
    prev: Node | null;
    next: Node | null;

    constructor(key: number, val: number) {
    
}

}

class LRUCache {

    private size: number;
    private capacity: number;
    private head: Node;
    private tail: Node;
    private cache: Map<number, Node>;

    constructor(capacity: number) {
    
}

    get(key: number): number {
    
}

    put(key: number, value: number): void {
    
}

    private removeNode(node: Node): void {
    
}

    private addToHead(node: Node): void {
    
}

}

/**
 * Your LRUCache object will be instantiated and called as such:
 * var obj = new LRUCache(capacity)
 * var param_1 = obj.get(key)
 * obj.put(key,value)
 */
